src/main/java/com/gluonhq/substrate/target/LinuxTargetConfiguration.java

```diff
@@ -77,9 +77,14 @@ public boolean link() throws IOException, InterruptedException {
 
     @Override
     List<String> getTargetSpecificLinkLibraries() {
-        List<String> defaultLinkFlags = new ArrayList<>(super.getTargetSpecificLinkLibraries());
-        defaultLinkFlags.addAll(Arrays.asList("-lextnet", "-lstdc++"));
-        return defaultLinkFlags;
+        List<String> linkFlags = new ArrayList<>();
+
+        linkFlags.add("-Wl,-Bstatic");
+        linkFlags.addAll(Arrays.asList("-ljava", "-lnio", "-lzip", "-lnet", "-ljvm", "-lstrictmath", "-lj2pkcs11", "-lsunec", "-lextnet"));
+        linkFlags.add("-Wl,-Bdynamic");
+        linkFlags.addAll(Arrays.asList("-lz", "-ldl", "-lstdc++"));
+
```

Comment:
Looks a little strange. All strings can be added at once on creation of the `List`. 
If the reason is clarity - just separating a groups of strings on different lines should suffice.